src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+          # interference to which the grants are entitled
+          if interference < fair_share:
+            interference_list[index] = interference
+            # Grant is satisfied
+            grant_initialize_flag[index] = True
+
+      for i, grant in enumerate(neighborhood_grants):
+
+        if grant_initialize_flag[i] is True:
+
+          # Remove satisfied CBSD grant from consideration
+          interference_quota = interference_quota - interference_list[i]
+
+          if num_grants > 0:
+            num_grants = num_grants - 1
+
+          # re-calculate the fair share if number of grants are not zero
+          if num_grants != 0:
+            fair_share = interference_quota / num_grants
+
+          # Get aggregate interference
+          aggr_interference += interference_list[i]
+
+          # Update EIRP and Interference only if managed grant
+          if grant.is_managed_grant:
+            if grant.grant_index not in post_iap_managed_grants:
+              post_iap_managed_grants[grant.grant_index] = {}
+
+              # Nested dictionary - contains dictionary of protection
+              # constraint over which grant extends
+              post_iap_grant = PostIAPGrantInfo(eirp=grants_eirp[i], 
+                                 interference=interference_list[i])
+              post_iap_managed_grants[grant.grant_index][channel_constraint] =\
+                                                             post_iap_grant
+        else:
+          grants_eirp[i] = grants_eirp[i] - 1
+
+      aggregate_interference[channel_constraint] = aggr_interference
+
+
+def performIapForEsc(protected_entity, grant_objects, iap_margin_esc):
+  """ Compute post IAP interference margin for ESC 
+  
+  IAP algorithm is run over ESC passband 3550-3660MHz for Category A CBSDs and 
+  3550-3680MHz for Category B CBSDs
+
+  Args:
+    protected_entity: List of protection points inside protection area
+    grant_objects: a list of grant requests, each one being a dictionary 
+                   containing grant information
+    iap_margin_esc: Pre IAP threshold value for ESC incumbent type(mW/RBW)
+  """
+
+  # Get the protection point of the ESC
+  esc_point = protected_entity['installationParam']
+  protection_point = (esc_point['latitude'], esc_point['longitude'])
+
+  # Get azimuth radiation pattern
+  ant_pattern = esc_point['azimuthRadiationPattern']
+
+  # Initialize array of size length of ant_pattern list
+  ant_pattern_gain = np.arange(len(ant_pattern))
+  for i, pattern in enumerate(ant_pattern):
+    ant_pattern_gain[i] = pattern['gain']
+
+  # Get ESC antenna information
+  esc_antenna_info = interf.EscInformation(antenna_height=esc_point['height'],
+                       antenna_azimuth=esc_point['antennaAzimuth'],
+                       antenna_gain=esc_point['antennaGain'],
+                       antenna_pattern_gain=ant_pattern_gain)
+
+  pool = Pool(processes=min(interf.NUM_OF_PROCESSES, len(protection_point)))
```

Comment:
What is the use of pool here? It is currently unused.
If you want to use pool, you have to either jobs to the pool, or use the pool.map() construct.

Reply:
Updated   to pool.map construct